src/geometry/contacts.rs

```diff
@@ -331,6 +341,13 @@ fn compute_contacts_for_pair_of_cells(
                     let (fluid_j, particle_j, is_boundary_j) = entry.into_tuple();
                     let pi = fluids[*fluid_i].positions[*particle_i];
                     let pj = if is_boundary_j {
+                        let bj = &boundaries[fluid_j];
+                        if !fluids[*fluid_i]
+                            .boundary_interaction
+                            .test(bj.fluid_interaction)
+                        {
+                            continue;
+                        }
                         boundaries[fluid_j].positions[particle_j]
                     } else {
                         fluids[fluid_j].positions[particle_j]
```

Comment:
There is a good chance that adding support for groups between two fluids is only a matter of adding a check in this branch too.